File: DataStructures/LinkedList/AddNodeAfterGiven.py
#This function is in the linked list class
#Inserts a new node after the given prev_node

def insertAfter(self, prev_node, new_data):

    #check if the given prev_node exists
    if prev_Node is None:
        print "The given previous node must be in LinkedList."
        return
    

    new_node = Node(new_data)
    new_node.next = prev_node.next
    prev_node.next = new_node

File: DataStructures/LinkedList/AddNodeAfterGiven.java
package DataStructures.LinkedList;
/* This function would be in the linkedlist class
It inserts a new node after a given previous node
*/
public void insertAfter(Node prev_node, int new_data)
{
    if(prev_node == null)
    {
        System.out.println("The given previous node cannot be null");
        return;
    }

    Node new_node = new Node(new_data);

    new_node.next = prev_node.next;

    prev_node.next = new_node;
}
